core/services/keeper/upkeep_executer_test.go

```diff
@@ -82,18 +83,52 @@ func Test_UpkeepExecutor_PerformsUpkeep_Happy(t *testing.T) {
 		require.Len(t, runs, 1)
 		_, ok := runs[0].Meta.Val.(map[string]interface{})["eth_tx_id"]
 		assert.True(t, ok)
+
+		ethMock.AssertExpectations(t)
 	})
 
-	t.Run("skips upkeep on non-triggering block number", func(t *testing.T) {
-		head := models.NewHead(big.NewInt(21), cltest.NewHash(), cltest.NewHash(), 1000)
+	t.Run("triggers exactly one upkeep if heads are skipped but later heads arrive within range", func(t *testing.T) {
+		store, ethMock, executor, registry, upkeep, job, jpv2 := setup(t)
+
+		registryMock := cltest.NewContractMockReceiver(t, ethMock, keeper.RegistryABI, registry.ContractAddress.Address())
+		registryMock.MockResponse("checkUpkeep", checkUpkeepResponse)
+
+		// turn falls somewhere between 20-39 (blockCountPerTurn=20)
+		// heads 20 thru 35 were skipped (e.g. due to node reboot)
+		head := models.NewHead(big.NewInt(36), cltest.NewHash(), cltest.NewHash(), 1000)
+
+		executor.OnNewLongestChain(context.Background(), head)
+		cltest.WaitForCount(t, store, models.EthTx{}, 1)
+		assertLastRunHeight(t, store, upkeep, 36)
+		runs := cltest.WaitForPipelineComplete(t, 0, job.ID, 1, jpv2.Jrm, time.Second, 100*time.Millisecond)
+		require.Len(t, runs, 1)
+		_, ok := runs[0].Meta.Val.(map[string]interface{})["eth_tx_id"]
+		assert.True(t, ok)
+
+		// heads 36, 37 etc do nothing
+		for i := 36; i < 40; i++ {
+			head = models.NewHead(big.NewInt(int64(i)), cltest.NewHash(), cltest.NewHash(), 1000)
+			executor.OnNewLongestChain(context.Background(), head)
+			cltest.AssertCountStays(t, store, models.EthTx{}, 1)
+		}
+
+		// head 40 triggers a new run
+		head = models.NewHead(big.NewInt(40), cltest.NewHash(), cltest.NewHash(), 1000)
```

Comment:
minor, but unless you need the timestamp, you could just use `cltest.Head(40)` for this and similar